f3/granite.go

```diff
@@ -460,6 +477,32 @@ func (i *instance) tryCommit(round uint32) {
 	}
 }
 
+// maybeAmplify implements Bracha amplification:
+// If a weak quorum of DECIDE messages has been received, at least one correct participant must have sent one
+// (and there exists a proof of it, albeit not necessarily locally present).
+// That means that this value must be terminated by every correct participant.
+// Broadcasting a DECIDE message is necessary for liveness if the broadcast primitive is only best-effort broadcast.
+// If broadcast() implemented reliable broadcast, this step would not be necessary.
+func (i *instance) maybeAmplify(value ECChain) {
+	weakQuorum := i.decision.HasWeakQuorumAgreement(value.HeadCIDOrZero())
+	if weakQuorum && i.phase != DECIDE && i.phase != TERMINATED {
+		i.value = value
+		i.beginDecide()
```

Comment:
I don't agree that receiving a DECIDE message should update the node's value nor cause a state transition into the DECIDE phase. The FIP PR does not do this either. This should be _only_ Bracha amplification.

Taking that as an assumption, the phase checks here are unwanted, but you will need some `decideSent` state to avoid a rebroadcast. The complexity of https://github.com/filecoin-project/go-f3/pull/30#discussion_r1445459148 will still be present, though. Could you please add a comment when decideSent is set true about the implications for the node arriving at its own decision from COMMITs?

Again, I hope we can just delete this after explicit validation.

Reply:
As a matter of fact. We can still keep DECIDE messages completely unjustified (not verifying any justification), and still remove the Bracha amplification. 

The reasoning is that eventually all (honest) nodes will receive enough COMMITs to send their decide, enter then the DECIDE phase, and then they can just check the locally delivered (unjustified) DECIDE messages to check for a strong quorum. 

Therefore, I have completely remove `i.decideSent` and `i.maybeAmplify`, but still left DECIDE with no justification whatsoever. Delivered DECIDE messages are immediately saved in quorumState but only when the node receives enough COMMITs to transit its state to DECIDE will it even check for the received DECIDE (and no Byzantine behavior changes anything because we check for a strong quorum of DECIDE).

Reply:
EDIT: We can forget about justifying DECIDE messages now that we have implicit justification and assume reliable broadcast. Without assuming reliable broadcast then we would need explicit justification and justify DECIDE messages (no bracha).